cpp/locality/NeighborBond.h

```diff
@@ -14,15 +19,16 @@ struct NeighborBond
     // For now, id = query_point_idx and ref_id = point_idx (into the NeighborQuery).
     constexpr NeighborBond() = default;
 
-    constexpr NeighborBond(unsigned int query_point_idx, unsigned int point_idx, float d = 0, float w = 1)
-        : query_point_idx(query_point_idx), point_idx(point_idx), distance(d), weight(w)
+    constexpr NeighborBond(unsigned int query_point_idx, unsigned int point_idx, float d, float w,
```

Comment:
Shouldn't distance be either be computed on the fly or in the constructor. I see no reason to pass it any more. This way we have to be careful to ensure that the vector is equivalent to the distance rather than being sure by the object's methods or construction.

Reply:
Good point! There is now an invariant to maintain in this struct. We should actually go a step further and convert it to a class with private members and getters/setters to maintain data consistency. I think I noticed this at one point too and forgot to go back and change this part.